### PR TITLE
Fix cancel button in the page of project edit not work

### PR DESCRIPTION
**Pull request by sillyguodong**
Before, in project edit page, the cancel button is not work.

https://user-images.githubusercontent.com/33891828/227182731-6478e29f-0e52-48c4-beb0-6a7d1dda6a1d.mov

1. The wrong classname `cancel` was added to the `<a>` tag. That classname caused the default click event of `<a>` tag to be cancelled. Because we have the following settings in the global. So I remove the classname `cancel`.

https://github.com/go-gitea/gitea/blob/9be90a58754061171bbd5025d85d2b891364efd3/web_src/js/features/common-global.js#L325-L327

2. Another change is that page will redirect to the previous page.

https://user-images.githubusercontent.com/33891828/227187326-c653c6d6-9715-440f-a732-ba0a6f012c81.mov

### CODE REVIEW

**Comment by wxiaoguang** [comment]
It's a regression of my suggestion in #23337

The proper fix should be : use `$(document).on('click', 'form button.ui.cancel.button', (e) => {` instead of `$(document).on('click', 'form .ui.cancel.button', (e) => {`

**Review by zeripath — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

File: templates/projects/new.tmpl [hunk 1]
```diff
@@ -50,7 +50,7 @@
 				<div class="ui divider"></div>
 				<div class="ui left">
 					{{if .PageIsEditProjects}}
-					<a class="ui cancel button" href="{{.RepoLink}}/projects">
+					<a class="ui cancel button" href="/{{.Org.Name}}/-/projects{{if eq .redirect "project"}}/{{.projectID}}{{end}}">
```
Comment (wxiaoguang) [inline templates/projects/new.tmpl]: Wait, this is incorrect.

The "AppSubUrl" is missing.

Comment (yp05327) [reply templates/projects/new.tmpl]: I think `$.HomeLink` is what you need. Same usage at line 6.

File: templates/projects/new.tmpl [hunk 2]
```diff
@@ -50,7 +50,7 @@
 				<div class="ui divider"></div>
 				<div class="ui left">
 					{{if .PageIsEditProjects}}
-					<a class="ui cancel button" href="{{.RepoLink}}/projects">
+					<a class="ui cancel button" href="{{AppSubUrl}}/{{.Org.Name}}/-/projects{{if eq .redirect "project"}}/{{.projectID}}{{end}}">
```
Comment (yp05327) [inline templates/projects/new.tmpl]:
```suggestion
					<a class="ui cancel button" href="{{$.HomeLink}}/-/projects{{if eq .redirect "project"}}/{{.projectID}}{{end}}">
```

**Pull request merged**